test/functional/test_framework/test_framework.py

```diff
@@ -158,6 +161,22 @@ def main(self):
             self.options.tmpdir = tempfile.mkdtemp(prefix=TMPDIR_PREFIX)
         self._start_logging()
 
+        # Seed the PRNG. Note that test runs are reproducible if and only if
+        # a single thread accesses the PRNG. For more information, see
+        # https://docs.python.org/3/library/random.html#notes-on-reproducibility.
+        # The network thread shouldn't access random. If we need to change the
+        # network thread to access randomness, it should instantiate its own
+        # random.Random object.
+        seed = self.options.randomseed
+
+        if self.options.randomseed is None:
+            seed = random.randrange(sys.maxsize)
+        else:
+            self.log.debug("User supplied random seed {}".format(self.options.randomseed))
```

Comment:
Just noted the log message change. Agreed, either "Using user-supplied" or "User supplied" may be better than "Using user supplied".

Reply:
Or maybe the local `seed` should be removed and the seed should be stored in the options member? That way it could be accessed by tests to (re-seed) to the inital seed?

Reply:
Interesting option as well, if re-seeding would be useful.

Reply:
oops. Good spot @jonatack - fixed!